packages/vesting-base/src/contract.rs

```diff
@@ -174,6 +178,24 @@ impl BaseVesting {
         }
     }
 
+    pub fn set_vesting_token(
+        &self,
+        deps: DepsMut,
+        _env: Env,
+        info: MessageInfo,
+        token: AssetInfo,
+    ) -> Result<Response, ContractError> {
+        let mut config = self.config.load(deps.storage)?;
+        if info.sender != config.owner && info.sender != config.token_info_manager {
```

Comment:
I don't want to do it so. This way we have only one chance to set them otherwise the contract is bricked

Reply:
actually we only need this whole update stuff because we cannot do it in genesis. so in an ideal world we'd set vesting token info at genesis which itself is only one attempt action. anyway we need to come to a conclusion here with @foxpy because he's done the oracle contract updatable in this way.

Reply:
having it immutable (if we set it at the genesis stage) or once mutable (as I suggested) improves contract's reliability for us as users imo. this way the system is protected from vesting token substitution and the rest of the asset information in tge contracts in the future by a fraud